src/Neo/SmartContract/Native/NativeContract.cs

```diff
@@ -158,41 +158,40 @@ protected NativeContract()
                 OrderBy(p => p.Order).ToList().AsReadOnly();
 
             // Calculate the initializations forks
-            usedHardforks =
-                methodDescriptors.Select(u => u.ActiveIn)
-                .Concat(methodDescriptors.Select(u => u.DeprecatedIn))
-                .Concat(eventsDescriptors.Select(u => u.ActiveIn))
-                .Concat(new Hardfork?[] { ActiveIn })
-                .Where(u => u is not null)
-                .OrderBy(u => (byte)u)
-                .Cast<Hardfork>().ToImmutableHashSet();
-
-            contractsList.Add(this);
-            contractsDictionary.Add(Hash, this);
+            _usedHardforks =
+                _methodDescriptors.Select(u => u.ActiveIn)
+                    .Concat(_methodDescriptors.Select(u => u.DeprecatedIn))
+                    .Concat(_eventsDescriptors != null ? _eventsDescriptors.Select(u => u.ActiveIn) : [])
```

Comment:
this should be always not null

Reply:
this is code, use ! or do null check, you need to do something.